lib/fast-boot.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * @license
+ * Copyright 2021 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This module sets up the `fast-boot2` module, including testing whether
+ * we have permissions over the cache file before even attempting to load
+ * fast boot.
+ * AVOID LOADING OTHER MODULES HERE AS MUCH AS POSSIBLE, AND KEEP ALL
+ * FUNCTIONS SYNCHRONOUS!
```

Comment:
I had thought that functions had to be synchronous in this file because the `await` keyword cannot be used in the module top level in `bin/balena`, but I could probably refactor `bin/balena` to have an `async` function (so that await can be used) and then this file could have async functions and I could actually try parallelising the execution of some code with `Promise.all()`. Converting the PR back to draft while I test this.